### ISSUE TITLE
NSWindow.WindowController return an NSObject instead of NSWindowController

### ISSUE THREAD

**Issue by danien** [xamarin/xamarin-macios#7975]
Is there a particular reason why the NSWindow.WindowController property uses an NSObject instead of an NSWindowController (as per Apple's [AppKit API](https://developer.apple.com/documentation/appkit/nswindow/1419092-windowcontroller))?

Is this intended?

```
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public virtual NSObject WindowController {
	[Export ("windowController")]
	get;
	[Export ("setWindowController:")]
	set;
}
```

## Info
Visual Studio for Mac 8.4.5 (build 19)
From Visual Studio's About window (Xamarin.Mac package in project reference shows version number as 0.0.0.0):
Xamarin.Mac
Version: 6.10.0.17 (Visual Studio Community)
Hash: 5f802ef53
Branch: xcode11.3
Build date: 2020-01-07 11:53:06-0500

**Comment by chamons** [comment]
Many of the most core API bindings in UIKit and AppKit were bound in a time where Apple's headers were significantly less descriptive. Many types returned id and did not list version info, for example.

Due to the "loose" nature of ObjC's type system and selector invocation in general, Apple was able update the return value to a more concrete type. However, C#'s type system will not allow that without a breaking API change.

The Xamarin.iOS and Mac team has a strict policy of avoiding API breaks whenever policy.

I've marked this issue as "XAMCORE_4_0", which is our internal description for a time when we are forced to make an API break and should fix everything that requires changes like this.

Thanks for the report.

**Comment by danien** [comment]
I see. Thanks for the explanation.

**Comment by rolfbjarne** [comment]
Fixed in 156ddead7b.
